code-challenges.test.js

```diff
@@ -10,37 +10,192 @@
+// Time:        0.533 s, estimated 1 s
+
 // b) Create the function that makes the test pass.
+ // Use filter to filter out anything out of the array that isn't a number & map to create a new aray out of the filtered array. With the new filtered array i'll divide each number by 3 and put it into a new array. 
+const arrayDivider = (array) => {
+  const newArray = array.filter(item =>typeof item === 'number');
+  const newArray2 = newArray.map(num => num %3 );
+  return newArray2
+};
+console.log(arrayDivider(hodgepodge1));
+console.log(arrayDivider(hodgepodge2));
+
+
+
+
+
 
-// --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
-// a) Create a test with an expect statement using the variables provided.
+
+
+
+
+//3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 const cubeAndSum1 = [2, 3, 4]
 // Expected output: 99
 const cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
 
+
+// a) Create a test with an expect statement using the variables provided.
+
+describe("arrayCuber",()=>{
+  it("takes in an array of numbers and returns the sum of all the numbers cubed.",()=>{
+    expect(cubeAndSum1).toEqual(99)
+    expet(cubeAndSum2).toEqual(1125)
+  })
+})
+
+
+
+// FAIL  ./code-challenges.test.js
+// arrayDivider
+//   ✓ takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3. (3 ms)
+// arrayCuber
+//   ✕ takes in an array of numbers and returns the sum of all the numbers cubed. (3 ms)
+
+// ● arrayCuber › takes in an array of numbers and returns the sum of all the numbers cubed.
+
+//   expect(received).toEqual(expected) // deep equality
+
+//   Expected: 99
+//   Received: [2, 3, 4]
+
+//     147 | describe("arrayCuber",()=>{
+//     148 |   it("takes in an array of numbers and returns the sum of all the numbers cubed.",()=>{
+//   > 149 |     expect(cubeAndSum1).toEqual(99)
+//         |                         ^
+//     150 |     expet(cubeAndSum2).toEqual(1125)
+//     151 |   })
+//     152 | })
+
+//     at Object.toEqual (code-challenges.test.js:149:25)
+
+// Test Suites: 1 failed, 1 total
+
+
+
 // b) Create the function that makes the test pass.
+
+
+
+// So to get cubeAndSum1 to equal 99... I had to google how a cubed number works. 
+//I should multiple each number in the array by 3 of itself so 2x2x2 =8, 3x3x3 = 27 and 4x4x4= 64  and then add all of them to the sum which should equal 99.
+// So I used a for loop so that it can just loop over each number and multiply each number by itself 3 times and add the sum into a new array. The starting sum should be 0 and at the end it'll be the sum of each number cubed and added up.
+
+//
+ const arrayCuber = (array)=>{
+  let cubedSum = 0;
+  for(let i = 0;i <array.length; i++){
+    cubedSum += array[i]**3;
+  } 
+  return cubedSum
+ }
+ console.log(arrayCuber(cubeAndSum1));
+ console.log(arrayCuber(cubeAndSum2));
```

Comment:
Excellent work! To step it to a higher level, try  refactoring using a higher order function instead of a for loop.  This works, though and is a great solution!  Do what you know!